provisioner/shell/provisioner.go

```diff
@@ -202,20 +210,38 @@ func (p *Provisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packe
 		scripts = append(scripts, tf.Name())
 
 		// Write our contents to it
-		writer := bufio.NewWriter(tf)
-		_, _ = writer.WriteString(fmt.Sprintf("#!%s\n", p.config.InlineShebang))
+		fileWriter := bufio.NewWriter(tf)
```

Comment:
I would suggest moving the logic that creates the file and its writer to after the loop that feeds the temporary buffer, so the file is only created if there are no problems with the command's creation.

Reply:
Come to think of it, do we really need a `bufio.NewWriter` here?
I would think we can use `tf` directly to write to the file, no need to have an extra buffer in between, we likely don't have enough data to write for it to make a noticeable difference, and we can remove a layer of indirection this way.